frontend/model/contracts/group.js

```diff
@@ -141,6 +142,46 @@ function isActionYoungerThanUser (actionMeta: Object, userProfile: ?Object): boo
   return compareISOTimestamps(actionMeta.createdDate, userProfile.joinedDate) > 0
 }
 
+function updateGroupStreaks ({ state, getters }) {
+  const {
+    groupIncomeAdjustedDistribution,
+    thisPeriodPaymentInfo
+  } = sbp('state/vuex/getters')
```

Comment:
Important to understand about contracts: they cannot access data outside of the contracts.

That is one immutable rule. Notice we're passing in `getters` via the function's parameters. Those are the only getters that are allowed.

Accessing `'state/vuex/getters'` is not allowed. If we had a way to detect data-access outside of the contracts, we would automate it as a hard error during linting. But we don't, so I'm acting as the linter. 😄 

The reason is because contracts must build up the same state consistently across clients. If we access data that is outside of the contracts, the state that will end up in the contracts will be inconsistent across different clients, and everything will explode.

As an example — `thisPeriodPaymentInfo` depends on `currentPaymentPeriod` Vuex getter, which in turn depends on `reactiveDate.date` — information that is different on each client. Therefore it cannot be used.

So - you can only access the getters that are defined on the contract - not the Vuex getters. The getters that are allowed are defined in this file and are accessible via the `getters` parameter that's being passed around.

Reply:
> Accessing `'state/vuex/getters'` is not allowed.

*To be clear: the only place where accessing `'state/vuex/getters'` is allowed in contracts, is in sideEffects, because those do not affect the state of contracts themselves.

Reply:
Thanks @taoeffect for a detailed explanation about this. I will keep the below in mind when changing any stuff related to the contract, moving forwards. :)

> Important to understand about contracts: they cannot access data outside of the contracts.

And the `updateGroupStreaks()` function has been updated so that it doesn't use vuex state for the logic.